SwiftSyntaxDevUtils/Sources/swift-syntax-dev-utils/commands/IntegrityCheck.swift

```diff
@@ -0,0 +1,56 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import ArgumentParser
+import Foundation
+
+struct IntegrityCheck: ParsableCommand {
+  static var configuration: CommandConfiguration {
+    return CommandConfiguration(
+      abstract: """
+        Ensure changes are fully tested, formatted, and validated before pull request submission.
+        """
+    )
+  }
+
+  @Flag(help: "Enable verbose logging.")
+  var verbose: Bool = false
+
+  func run() throws {
+    try runCommand("format")
```

Comment:
Instead of spawning new processes, I think we should just execute the commands in the current process. The following code does this, for example.

https://github.com/apple/swift-syntax/blob/098b177830c0d2e614b3757241b2572d758f9143/SwiftParserCLI/Sources/swift-parser-cli/Commands/Reduce.swift#L117-L129

Reply:
If I'm not mistaken, this code does not execute the commands. The `run` function in this instance is simply a static function that isn't derived from the `ParsableCommand` protocol. It's actually a custom function within the `VerifyRoundTrip` struct that encapsulates all the logic for the command.

We can not run the command directly, can we? 

If we were to adhere to the `Single Responsibility Principle`, we would need to refactor all the commands so that they have one responsibility—acting as an interface for command creation—and then move the entire business logic out of them. However, the question remains: given that this is merely a developer utility package, is it worth making these changes?

Reply:
I think what we should be able to do is to either add a similar static `run` function to those commands as well (like we do with `VerifyRoundTrip`) or add a constructor to the commands that allows you to set all the command line arguments programatically.

Reply:
Let's discus it further here: #2350